charts/policy-reporter/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.ui.networkPolicy.enabled -}}
```

Comment:
this NetworkPolicy should be applied independent of the UI. Because you still need it without the PolicyReporter UI to access the Metrics API and to enable egress traffic to targets like Loki

So you should also add a separat networkPolicy value for the main chart.